ldapwatch.go

```diff
@@ -128,19 +121,8 @@ func (w *Watch) stop() {
 }
 
 // perform the search and check the results with the Checker
-func (w *Watch) tick() error {
-	var result Result
-	sr, err := w.search()
-
-	if err != nil {
-		result = Result{Watch: w, Err: err}
-	} else {
-		result = Result{Watch: w, Results: sr}
-	}
-
-	w.checker.Check(result)
-
-	return nil
+func (w *Watch) tick() {
+	w.checker.Check(w.search())
```

Comment:
This is now laughably trivial (and could arguably be inlined into `ldapwatch.Watch.start()`).

Should we be checking for `error`s and handling them some how here? What would we do?

Currently, this entirely relies on the `Checker` to handle the `error` as it see fits. I think this is ideal (though I'm not sure the `Checker` has any way to halt the worker other than a panic).